src/main/kotlin/org/rust/lang/core/psi/ext/RsAttr.kt

```diff
@@ -20,3 +22,14 @@ val RsAttr.owner: RsDocAndAttributeOwner?
         // Throw exception so that any problems with this property aren't silent and are instead easily findable
         else -> error("Unsupported attribute type: $this")
     }
+
+fun RsAttr.isBuiltinWithName(target: String): Boolean {
+    return if (metaItem.name != target) false
+    else {
+        val name = metaItem.name ?: return false
+
+        if (name !in STD_ATTRIBUTES) false
+        else
+            !hasInScope(name, MACROS)
+    }
+}
```

Comment:
nit: I think we can slightly rewrite it to improve readability
```kotlin
fun RsAttr.isBuiltinWithName(target: String): Boolean {
    val name = metaItem.name ?: return false
    if (name != target) return false
    if (name !in STD_ATTRIBUTES) return false
    return !hasInScope(name, MACROS)
}
```

Also, I think we should move it into `RsErrorAnnotator.kt` file and make it private - unlikely we need it somewhere else

Reply:
Done.